src/v/cluster/rm_stm.cc

```diff
@@ -1813,7 +1824,25 @@ ss::future<raft::stm_snapshot> rm_stm::do_take_local_snapshot(
         }
     }
     auto snapshot_offset = last_applied_offset();
+    // return units and execute heavy i/o operations
     apply_units.return_all();
+    co_await ss::max_concurrent_for_each(
+      std::move(aborted_snapshots), 64, [this](abort_snapshot& snapshot) {
+          return save_abort_snapshot(std::move(snapshot));
+      });
```

Comment:
Doesn't this have correctness issues in the aborted transactions read path.. here we are updating abort indexes before there is a snapshot file on disk..

If there is a call to fetch aborted_transactions() after a scheduling point before L1829 and it attempts to load a index file by looking up the abort_indexes list, it may not see it on disk?

Reply:
this is true, previously the code was vulnerable to the same issue, i am going to fix that. Great catch

Reply:
done